File: Ex-19/c++/Mensagem.cpp
#include "Mensagem.hpp"

Mensagem::Mensagem() : texto("")
{
}

Mensagem::~Mensagem()
{
}

void Mensagem::addMensagem(const string &texto)
{
  this->texto = texto;
}

const string &Mensagem::getMensagem() const
{
  return this->texto;
}

File: Ex-19/ex_19_java/Mensagem.java
public class Mensagem {
    private String texto = ""; // Inicializado para permitir a concatenação

    /**
     * Adiciona (concatena) um novo trecho de texto à mensagem existente.
     */
    public void addMensagem(String textoAdicional) {
        this.texto += textoAdicional;
    }

    /**
     * Retorna o texto completo da mensagem.
     */
    public String getMensagem() {
        return this.texto;
    }
}